Add -fexceptions to quantized kernel generated libs

Daksh-Shami opened the pull request:
Summary:
Added compiling with `-fexceptions` since quantized kernels have exception based code in them. Example error:
```
buck-out/ABC/gen/fbsource/6e53edb0a9a0d828/xplat/executorch/kernels/quantized/__generated_lib_combined__/out/RegisterCodegenUnboxedKernelsEverything.cpp:77:44: error: exception handling disabled, use '-fexceptions' to enable
   77 |             } catch (const std::exception& ex) {
      |           
```
This means the generated code uses exceptions in its code, so we should explicitly flag it to the compiler via the relevant BUCK targets.

Differential Revision: D84284541

swolchok reviewed (changes requested):
Review automatically exported from Phabricator review in Meta.

swolchok reviewed (approved):
Review automatically exported from Phabricator review in Meta.

Daksh-Shami committed:
Add exception free quantized kernels (#14962)

Summary:

Explicitly declaring generated libs with exception code off for quantized kernels as executorch does not use exceptions in general, and it can cause downstream errors. Detailed error is P1986979595 but the relevant part is:
```
buck-out/ABC/gen/fbsource/6e53edb0a9a0d828/xplat/executorch/kernels/quantized/__generated_lib_combined__/out/RegisterCodegenUnboxedKernelsEverything.cpp:77:44: error: exception handling disabled, use '-fexceptions' to enable
   77 |             } catch (const std::exception& ex) {
      |
```
This means the existing generated code uses exceptions in its code, so when we use these kernels with -fno-exceptions downstream, the build fails.

After this diff, we can use the exception free kernels with 'no_exceptions' suffix -- 
`//xplat/executorch/kernels/quantized:generated_lib_no_exceptions` or 
`//xplat/executorch/kernels/quantized:generated_lib_aten_no_exceptions` as appropriate. 

We still have `//xplat/executorch/kernels/quantized:generated_lib` and `//xplat/executorch/kernels/quantized:generated_lib_aten` available, same as before, so no downstream side effects are expected.

Reviewed By: swolchok

Differential Revision: D84284541
